### PR TITLE
Add E2E test using Broker, Trigger, and KafkaSink

### LINKED ISSUE #164: Create an E2E test using Brokers, Triggers and KafkaSinks
**Problem**

We should create a test with all the objects we manage. This can prevent
regressions or raise bugs in the implementation or with the configurations 
(eg RBAC config).

**Test Plan**

```
+---------+     +--------+     +---------+   +---------------------------+
|  Broker +---->+ Trigger+---->+KafkaSink+-->+kafka consumer (test image)|
+----+----+     +--------+     +----+----+   +---------------------------+
     |                              ^
     |          +--------+          |
     +--------->+ Trigger+----------+
                +--------+
```
(Use this diagram as test documentation)

- Send 10 events to the Broker and verify them using the `kafka-consumer` test image.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
`*`

**Exit Criteria**
An E2E test using Brokers, Triggers and KafkaSinks

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1

**Additional context (optional)**
Add any other context about the feature request here.

- the `/test` directory has some examples.

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes #164 

## Test Plan

```
+---------+     +--------+     +---------+   +---------------------------+
|  Broker +---->+ Trigger+---->+KafkaSink+-->+kafka consumer (test image)|
+----+----+     +--------+     +----+----+   +---------------------------+
     |                              ^
     |          +--------+          |
     +--------->+ Trigger+----------+
                +--------+
```

## Proposed Changes

- Add E2E test using Broker, Trigger, and KafkaSink

### CODE REVIEW

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/broker_trigger_sink_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
```
Comment (mattmoor) [inline test/e2e/broker_trigger_sink_test.go:20]: Format Go code:
```suggestion

import (
	"context"
	"strings"
	"testing"

	"github.com/stretchr/testify/assert"
	"k8s.io/apimachinery/pkg/types"
	"k8s.io/utils/pointer"
	eventingv1alpha1 "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/eventing/v1alpha1"
	eventingv1alpha1clientset "knative.dev/eventing-kafka-broker/control-plane/pkg/client/clientset/versioned/typed/eventing/v1alpha1"
	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/kafka"
	"knative.dev/eventing-kafka-broker/test/pkg/addressable"
	kafkatest "knative.dev/eventing-kafka-broker/test/pkg/kafka"
	"knative.dev/eventing-kafka-broker/test/pkg/sink"
	pkgtesting "knative.dev/eventing-kafka-broker/test/pkg/testing"
	testlib "knative.dev/eventing/test/lib"
	"knative.dev/eventing/test/lib/resources"
	duckv1 "knative.dev/pkg/apis/duck/v1"
)

```

File: test/e2e/broker_trigger_sink_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+/*
++---------+     +--------+     +---------+   +---------------------------+
+|  Broker +---->+ Trigger+---->+KafkaSink+-->+kafka consumer (test image)|
++----+----+     +--------+     +----+----+   +---------------------------+
+     |                              ^
+     |          +--------+          |
+     +--------->+ Trigger+----------+
+                +--------+
+ */
```
Comment (mattmoor) [inline test/e2e/broker_trigger_sink_test.go:29]: Format Go code:
```suggestion
*/
```

**Comment by pierDipi** [comment]
```
[ERROR] Failed to execute goal on project receiver: Could not resolve dependencies for project dev.knative.eventing.kafka.broker:receiver:jar:1.0-SNAPSHOT: Could not transfer artifact com.github.luben:zstd-jni:jar:1.4.4-7 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/com/github/luben/zstd-jni/1.4.4-7/zstd-jni-1.4.4-7.jar: Connection reset -> [Help 1]
```
Maven central
/retest

**Comment by pierDipi** [comment]
This test worked, flaky not
/retest

**Comment by pierDipi** [comment]
Flaky...
/retest

File: test/pkg/addressable/send.go [hunk 3]
```diff
@@ -60,11 +64,16 @@ func SendN(t *testing.T, n int, addressable Addressable) []string {
...
 				event.SetID(id)
 
+				for _, mutator := range mutators {
+					mutator(&event)
+				}
+
 				client.Namespace = addressable.Namespace
 
-				client.SendEventToAddressable(ctx, fmt.Sprintf("%s-%d", addressable.Name, i), addressable.Name, &addressable.TypeMeta, event)
+				name := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-%d", addressable.Name, i))
```
Comment (slinkydeveloper) [inline test/pkg/addressable/send.go:73]: I think you don't need to generate a name bringing in this dep because SendEventToAddressable takes care of it

Comment (pierDipi) [reply test/pkg/addressable/send.go:73]: No, it doesn't.

Comment (slinkydeveloper) [reply test/pkg/addressable/send.go:73]: Uhm my bad :smile:

**Pull request merged**